scalameter-core/src/main/scala/org/scalameter/MeasureBuilder.scala

```diff
@@ -38,12 +43,12 @@ class MeasureBuilder[T, U](
   
       if (ctx(Key.exec.requireGC)) compat.Platform.collectGarbage()
   
-      val measurements = measurer.measure[T, U](
+      val measurements = measurer.measure[T](
         ctx, ctx(Key.exec.benchRuns),
         setup, teardown, regen, b
       )
   
-      resultFunction(measurements)
+      resultFunction(measurements.map(_.value))
```

Comment:
Hm, I've dropped `Quantity` to tuple right after call to the `measurer.measure` and everything got serialized and deserialized properly.
So I think that this would be better workaround, but I still don't know the exact cause and it's weird to me that tuples work but custom `Quantity` class not.
Btw, I've tried to drop usage of `case class` here and mixing `Serializable` manually but that didn't work too.